### PR TITLE
Add prometheus agent mode for SKE

### PR DESCRIPTION
**Pull request by SimonKienzler**
Includes removal of charts for `seed-prometheus` and `aggregate-prometheus`.

**How to categorize this PR?**

/area TODO
/kind TODO

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator

```

### CODE REVIEW

File: pkg/operation/botanist/monitoring.go [hunk 1]
```diff
@@ -369,6 +377,29 @@ func (b *Botanist) DeploySeedMonitoring(ctx context.Context) error {
 		return err
 	}
 
+	// depending on the agentmode setting, prometheus is
+	// either deployed as a StatefulSet or as a Deployment.
+	// The respective other type needs to be removed, if existent
+	if b.HasPrometheusAgentModeSetting(ctx) && *b.Config.Monitoring.Shoot.AgentMode.Enabled {
```
Comment (nschad) [inline pkg/operation/botanist/monitoring.go:383]: Isn't `HasPrometheusAgentModeSetting` the same as the AgentMode.Enabled?

File: pkg/operation/botanist/monitoring.go [hunk 2]
```diff
@@ -740,5 +765,32 @@ func (b *Botanist) DeleteSeedMonitoring(ctx context.Context) error {
 		},
 	}
 
+	// depending on the agentmode setting, prometheus is
+	// either deployed as a StatefulSet or as a Deployment
+	if b.HasPrometheusAgentModeSetting(ctx) && *b.Config.Monitoring.Shoot.AgentMode.Enabled {
```
Comment (nschad) [inline pkg/operation/botanist/monitoring.go:770]: same as above
```suggestion
	if b.HasPrometheusAgentModeSetting(ctx) {
```

File: pkg/operation/botanist/monitoring.go (hunk 1 above)
Comment (nschad) [inline pkg/operation/botanist/monitoring.go:383]:
```suggestion
	if b.HasPrometheusAgentModeSetting(ctx) {
```

File: pkg/operation/botanist/monitoring.go [hunk 3]
```diff
@@ -740,5 +765,32 @@ func (b *Botanist) DeleteSeedMonitoring(ctx context.Context) error {
...
+			},
+		})
+	}
+
 	return kutil.DeleteObjects(ctx, b.K8sSeedClient.Client(), objects...)
 }
+
+// If agentMode is enabled, Prometheus is deployed as a Deployment. Otherwise
+// it's a StatefulSet. This function allows easy switching between the two.
+func (b *Botanist) HasPrometheusAgentModeSetting(ctx context.Context) bool {
```
Comment (nschad) [inline pkg/operation/botanist/monitoring.go:791]: This only checks if the config option exists. For simplicity it should also check if AgentMode is enabled?

**Review by nschad — APPROVED** [review]

**Review by brumhard — APPROVED** [review]

**Review by dergeberl — APPROVED** [review]

**Pull request merged**